### ISSUE TITLE
Add check_date when creating or editing POIs

### ISSUE THREAD

**Issue by matheusgomesms** [organicmaps/organicmaps#7882]
Just as title says, a first iteration of having updated POIs in OSM (and OM, consequently) is adding the `check_date` tag when creating or updating POIs. Already discussed in #6811 and #7036.

What I'm proposing here is to simply add `check_date` when a POI is created or edited. Nothing fancy, nothing visible to the user (yet), no Place Page modifications, nothing. The user would never know (at this point) this tag was inserted.

Updating OSM API is really not an option (latest version from 2009, [with some minor updates since](https://wiki.openstreetmap.org/wiki/API_v0.6)).

This tag has a good support already (EveryDoor and StreetComplete, AFAIK), and could be a first step in using this information here in the future. Additionally, can benefit A LOT OSM mappers.

**Comment by biodranik** [comment]
What is the value of updating that tag for all newly created features? The default check date defaulting to the date of adding the feature means exactly the same, right?

**Comment by matheusgomesms** [comment]
It's easier to use `check_date`, and some editors already are using that tag.

**Comment by map-per** [comment]
A problem I see here:
If someone adds a place some time after visiting it this approach would add a wrong check_date (The day when the edit was made would be added and not the day when the user checked the place).

Isn't StreetComplete also able to check how much time passed since the last modification of an object?

**Comment by matheusgomesms** [comment]
> A problem I see here: If someone adds a place some time after visiting it this approach would add a wrong check_date (The day when the edit was made would be added and not the day when the user checked the place).

I agree that the tag might not reflect the exact date the feature was surveyed, but being OM mainly a mobile app, shouldn't this not happen frequently? I mean, one could armchair map using OM, but maybe that's unlikely? And even if one does that, this probably could be only a matter of days difference, so it doesn't change the overall picture (eg. very likely the POI would still be there)?

> Isn't StreetComplete also able to check how much time passed since the last modification of an object?

Don't know, actually.

**Comment by map-per** [comment]
> And even if one does that, this probably could be only a matter of days difference

But it's still wrong data and adding wrong data to OSM is not good.

**Comment by matheusgomesms** [comment]
> > And even if one does that, this probably could be only a matter of days difference
> 
> But it's still wrong data and adding wrong data to OSM is not good.

But how can any OSM editor prevent that? I can open OM right now and delete a POI, the check_date tag doesn't change that.

OM is actually for years adding wrong data in OSM (as your PR #6257 is trying to fix) and it's still not fixed.

Realistically speaking, in a scenario where this would be implemented, what is the % where this could happen (POI closed in real life before mapper editing in OM)? Let's remember, OM blocks editing when a new map release is out, so this editing window is around 1 month.

**Comment by matheusgomesms** [comment]
The advantage of a mobile app is that we can assume the mapper is looking at the POI, and the modification could be "truer" than any desktop editing. check_date here is more reliable than in iD, for example, even with the offline edits "issue".

**Comment by map-per** [comment]
- ID doesn't automatically add `check_date`, users have to manually add it. 
- And StreetComplete checks your GPS position to ensure that the you really survey the quests on site. If you are to far away from a quest or the GPS is turned off you have to confirm that you only added data surveyed on site.
- And in EveryDoor setting `check_date` is a manual step as well.

**Comment by matheusgomesms** [comment]
> * ID doesn't automatically add `check_date`, users have to manually add it.

 And I agree with that, iD can't know if you were there or not.

> * And in EveryDoor setting `check_date` is a manual step as well.

Is it? I just checked it, as soon you add a POI, the check_date is inserted. If you update a POI, check_date is inserted or updated (if there was another already).

About StreetComplete, I don't have it so I can't say much, although this functionality seems reasonable for me (and I believe will never be implemented in OM, due to complexity and focus of the app).

**Comment by map-per** [comment]
I don't think we need `check_date` for #6811 and #7036. We can also fall back to the date of last modification. This timestamp is stored in the osm database.

**Comment by map-per** [comment]
> Is it? I just checked it, as soon you add a POI, the check_date is inserted. If you update a POI, check_date is inserted or updated (if there was another already).

Ok, then I confused that with the option to check the existence of a place.

**Comment by matheusgomesms** [comment]
> I don't think we need `check_date` for #6811 and #7036. We can also fall back to the date of last modification. This timestamp is stored in the osm database.

That's the whole point of this opened issue: it seems (please any dev confirm this), that it is much easier to obtain check_date than the changeset timestamp. Also, check_date is more reliable than timestamp, to confirm that feature exists. Let's consider a case:

1 - a mapper is fixing alignment in a given area, using iD or JOSM. He/she doesn't delete or add anything, just fixes roads, buildings and POI alignment, using better imagery/GPS traces.
2 - a POI mapped 10 years ago (untouched), is now touched by that mapper. Therefore, the POI timestamp is updated.
3 - that POI was closed 5 years ago :)
4 - if any OSM editor relies on that info, it will think that the POI exists, just because the timestamp was updated.

I am not giving an absurd example, this alignment thing happens all the time in OSM. check_date is useful exactly for that (among other uses). The mapper is saying "This exists as of this date". More than [1 million uses already](https://taginfo.openstreetmap.org/keys/check_date#overview).

**Comment by matheusgomesms** [comment]
> The default check date defaulting to the date of adding the feature means exactly the same, right?

No, please see my previous answer.

**Comment by map-per** [comment]
Yes `check_date` is, of course, more reliable.

And just to avoid any misunderstandings: I'm not against showing `check_date` on the map or adding a "I confirm that this place is still there" editing functionality. I'm just against automatically adding tags based on assumptions that are not always. Because if we do that `check_date` will not be reliable any more.

**Comment by matheusgomesms** [comment]
Of course, no one wants bad data, I understand you. But I think the pros are way bigger than the possible cons. I still think that the number of people messing with data, regarding check_date, would be very very minimal.

**Comment by patepelo** [comment]
What if we only add check_date if the POI has been edited (not new) AND the user is near that location? 

It would be a start, and maybe we can give it a thought for the future to implement it for new POIs.

**Comment by biodranik** [comment]
Focusing on the root case may help: the check_date itself doesn't solve the problem of outdated POIs. Encouraging users to update POIs do.

**Comment by matheusgomesms** [comment]
> What if we only add check_date if the POI has been edited (not new) AND the user is near that location?

Seems a very good idea, not sure if it makes it harder to implement.

> Focusing on the root case may help: the check_date itself doesn't solve the problem of outdated POIs. Encouraging users to update POIs do.

This could be a first step towards #7036 :) Also, using check_date, now, could enable OSM mappers using EveryDoor or other editors to update POIs, while #7036 is not fully solved. Win-win situation.

**Comment by eisams** [comment]
> Focusing on the root case may help: the check_date itself doesn't solve the problem of outdated POIs. Encouraging users to update POIs do.

Indirectly it does

If users try out OSM/related apps and go to a POI and find out it is not there, they may be disappointed and move back to Google/Apple Maps which is more reliable today

By surfacing this tag, the users will be less disappointed and stick around. If they stick around, they may become contributors

It's also a big problem for corporate users that they can't reason about the age of the POIs, see this post by a former Meta employee that worked on their Places dataset that ended up in Overture: https://mastodon.social/@migurski/110799810490001999

If the tag gets more usage, more corporate users may be interested in using and contributing to the dataset (like they do with roads)

**Comment by biodranik** [comment]
Please define "by surfacing this tag to users". How exactly will it be "surfaced"?

**Comment by eisams** [comment]
> Please define "by surfacing this tag to users". How exactly will it be "surfaced"?

If the related https://github.com/organicmaps/organicmaps/issues/7036 is implemented, then the tag has been surfaced to users :)
